VCs/CategoryVC.swift

```diff
@@ -0,0 +1,90 @@
+//
+//  CategoryVC.swift
+//  To-Do App
+//
+//  Created by Ananth Desai on 18/02/22.
+//
+
+import UIKit
+
+class CategoryVC: UIViewController, UITableViewDelegate, UITableViewDataSource {
+    var selectedIndex: Int = 0
+    var categoryTableView: UITableView!
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        // Do any additional setup after loading the view
+        let constraints = setupCategoryTableView()
+        NSLayoutConstraint.activate(constraints)
+
+        navigationItem.rightBarButtonItem = UIBarButtonItem(
+            title: navbarRightButtonTitle,
+            style: .plain,
+            target: self,
+            action: #selector(didTapAddCategory)
+        )
+    }
+
+    override func viewDidLayoutSubviews() {
+        super.viewDidLayoutSubviews()
+        categoryTableView.frame = view.bounds
+    }
+
+    func setupCategoryTableView() -> [NSLayoutConstraint] {
+        categoryTableView = UITableView()
+        categoryTableView.register(UITableViewCell.self, forCellReuseIdentifier: "Cell")
+        categoryTableView.translatesAutoresizingMaskIntoConstraints = false
+        view.addSubview(categoryTableView)
+        categoryTableView.dataSource = self
+        categoryTableView.delegate = self
+
+        return [
+            categoryTableView.centerYAnchor.constraint(equalTo: view.centerYAnchor),
+            categoryTableView.leftAnchor.constraint(equalTo: view.leftAnchor),
+            categoryTableView.rightAnchor.constraint(equalTo: view.rightAnchor),
+            categoryTableView.topAnchor.constraint(
+                equalTo: view.topAnchor, constant: categoryTableViewTopAnchorConstant
+            ),
+        ]
+    }
+
+    @objc func didTapAddCategory() {
+        let alert = UIAlertController(title: alertTitle, message: "", preferredStyle: .alert)
+        alert.addTextField()
+        alert.addAction(UIAlertAction(title: alertLeftButtonTitle, style: .cancel, handler: nil))
+        alert.addAction(UIAlertAction(title: alertRightButtonTitle, style: .default, handler: { [alert] _ in
+            sourceData.append(Item(category: alert.textFields![0].text!, array: []))
+            self.categoryTableView.reloadData()
+        }))
+        present(alert, animated: true, completion: nil)
+    }
+}
+
+extension CategoryVC {
```

Comment:
You can remove the confirmation to protocol in class definition and add it here
```suggestion
extension CategoryVC: UITableViewDelegate, UITableViewDataSource {
```